demos/TransferLearning/TransferLearningDemo.py

```diff
@@ -0,0 +1,481 @@
+        self.model.compile(optimizer=self.sgd,
+                           loss=self.OBJECTIVE_FUNCTION,
+                           metrics=self.LOSS_METRICS)
+        predictions_initial = self.model.predict_generator(self.test_generator,
+                                                           verbose=0,
+                                                           callbacks=[ProgressBar(self)])
+        self.predicted_class_indices_init = np.argmax(predictions_initial, axis=1)
+
+        # ~ operator is equivalent to -x-1; so 0 becomes -1, 1 becomes -2; add 2 to implement xnor
+        guesses = ~(self.predicted_class_indices_init ^ self.test_class_indices) + 2
+
+        # Randomly select 9 images that guessed incorrectly
+        self.wrong_guesses = []
+        while len(self.wrong_guesses) < 9:
+            randomIndex = random.randint(0, len(self.predicted_class_indices_init) - 1)
+            if not guesses[randomIndex]:
+                self.wrong_guesses.append(randomIndex)
+
+        with self.out_initial:
+            print("Foo")
+            f, ax = plt.subplots(3, 3, figsize=(15, 15))
+
+            for i in range(0, 9):
+                test_image = os.path.join(self.test_dir,
+                                          self.test_generator.filenames[self.wrong_guesses[i]])
+                imgRGB = mpimg.imread(test_image)
+
+                predicted_class = "Dog" if self.predicted_class_indices_init[
+                    self.wrong_guesses[i]] else "Cat"
+
+                ax[i // 3, i % 3].imshow(imgRGB)
+                ax[i // 3, i % 3].axis('off')
+                ax[i // 3, i % 3].set_title("Predicted:{}".format(predicted_class), color='r')
+
+                if predicted_class.lower() in self.test_generator.filenames[self.wrong_guesses[i]]:
+                    ax[i // 3, i % 3].set_title("Predicted:{}".format(predicted_class), color='g')
+            plt.show()
+            print("bar")
+
+    def on_model_change(self, change):
+        if change['type'] == 'change' and change['name'] == 'value':
+            self.classify_button.disabled = True
+
+            # Models
+            if change['new'] == 'ResNet50':
+                self.epoch_slider.min = 1
+                self.epoch_slider.max = 16
+                self.epoch_slider.value = 2
+            if change['new'] == 'MobileNet v2':
+                self.epoch_slider.min = 2
+                self.epoch_slider.max = 50
+                self.epoch_slider.value = 10
+
+    def on_ngraph_change(self, change):
+        if change['type'] == 'change' and change['name'] == 'value':
+            i = self.ngraph_backends.index(change['new'])
+
+            if self.ngraph_backends[i] == 'DISABLED':
+                self.use_ngraph = False
+                ngraph_bridge.disable()
+            else:
+                self.use_ngraph = True
+                ngraph_bridge.enable()
+                ngraph_bridge.set_backend(self.ngraph_backends[i])
+
+    def init_gui(self):
```

Comment:
If this is a command-line version of the demo, does the gui stuff still work? Is this function still needed/relevant?

Reply:
This is intended for use in both a notebook as well as CI integration for nGraph. The gui call won't actually do anything if testing is enabled.